katana/wui/core/static/core/js/main.js

```diff
@@ -836,6 +836,150 @@ var katana = {
+                                                   '</div>' +
+                                                   '<div class="file-explorer-body">' +
+                                                       '<div class="up-btn-bar">' +
+                                                            '<button id="explorer-up" class="btn btn-info">...</button>' +
+                                                       '</div>' +
+                                                       '<div class="directory-data-div to-scroll">' +
+                                                           '<div id="directory-data" class="full-size">' +
+                                                           '</div>' +
+                                                       '</div>' +
+                                                   '</div>' +
+                                                   '<div class="file-explorer-footer">' +
+                                                       '<div class="full-size text-center">' +
+                                                            '<button id="explorer-accept" class="btn btn-success">OK</button>' +
+                                                            '<button id="explorer-dismiss" class="btn btn-success">Cancel</button>' +
+                                                       '</div>' +
+                                                   '</div>' +
+                                              '</div>' +
+                                         '</div>'
+                var $currentPage = katana.$activeTab;
+                var $tabContent = $currentPage.find('.page-content-inner');
+                $(explorer_modal_html).prependTo($tabContent);
+
+                $directoryData = $tabContent.find('#directory-data');
+
+                $directoryData.jstree({
+                    "core": {
+                        "data": [data]
+                    }
+                });
+
+                $directoryData.jstree().hide_dots();
+
+                $tabContent.find('#explorer-accept').on('click', function(){
+
+                    katana.acceptFileExplorer(callBack_on_accept);
+                });
+
+                $tabContent.find('#explorer-dismiss').on('click', function(){
+
+                    katana.dismissFileExplorer(callBack_on_dismiss);
+                });
+
+                $tabContent.find('#explorer-up').on('click', function(){
+
+                    katana.upFileExplorer(data.li_attr["data-path"]);
+                });
+
+            });
+
+	},
+
+	acceptFileExplorer: function(callBack){
+	    var $currentPage = katana.$activeTab;
+	    $fileExplorerElement = $currentPage.find('div .overlay');
+
+	    $selectedValue = $fileExplorerElement.find('[aria-selected=true]')
+
+	    var data_path = $selectedValue.attr("data-path");
+
+	    if(data_path == undefined){
+	        alert("Nothing selected");
+	        return;
+	    }
+
+	    $fileExplorerElement.remove();
+
+	    if(callBack){
```

Comment:
callBack && callBack(data_path) is the correct way to do this.

Reply:
Addressed